In [1]:
# read the packages
# Read the data
# cat and numerical

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

**Read the data**

In [3]:
visa_df=pd.read_csv("Visadataset.csv")
visa_df.head()

case_id continent education_of_employee has_job_experience  \
0  EZYV01      Asia           High School                  N   
1  EZYV02      Asia              Master's                  Y   
2  EZYV03      Asia            Bachelor's                  N   
3  EZYV04      Asia            Bachelor's                  N   
4  EZYV05    Africa              Master's                  Y   

  requires_job_training  no_of_employees  yr_of_estab region_of_employment  \
0                     N            14513         2007                 West   
1                     N             2412         2002            Northeast   
2                     Y            44444         2008                 West   
3                     N               98         1897                 West   
4                     N             1082         2005                South   

   prevailing_wage unit_of_wage full_time_position case_status  
0         592.2029         Hour                  Y      Denied  
1       83425.6500         Year                  Y   Certified  
2      122996.8600         Year                  Y      Denied  
3       83434.0300         Year                  Y      Denied  
4      149907.3900         Year                  Y   Certified

**Divide into categorical and numerical columns**

In [4]:
types=dict(visa_df.dtypes)
cat_columns=[key for key,value in types.items() if value=='object']
num_columns=[key for key ,value in types.items() if value!='object']

In [5]:
cat_columns=visa_df.select_dtypes(include='object').columns
num_columns=visa_df.select_dtypes(exclude='object').columns
cat_columns

Index(['case_id', 'continent', 'education_of_employee', 'has_job_experience',
       'requires_job_training', 'region_of_employment', 'unit_of_wage',
       'full_time_position', 'case_status'],
      dtype='object')

**how to select a specific columns**

In [6]:
visa_df['continent'] # series type

0          Asia
1          Asia
2          Asia
3          Asia
4        Africa
          ...  
25475      Asia
25476      Asia
25477      Asia
25478      Asia
25479      Asia
Name: continent, Length: 25480, dtype: object

In [7]:
visa_df.loc[:,['continent']]

continent
0          Asia
1          Asia
2          Asia
3          Asia
4        Africa
...         ...
25475      Asia
25476      Asia
25477      Asia
25478      Asia
25479      Asia

[25480 rows x 1 columns]

In [8]:
visa_df[['continent']]

continent
0          Asia
1          Asia
2          Asia
3          Asia
4        Africa
...         ...
25475      Asia
25476      Asia
25477      Asia
25478      Asia
25479      Asia

[25480 rows x 1 columns]

In [9]:
visa_df['continent'] # series type
visa_df[['continent']] # dataframe
visa_df.continent # series
visa_df.loc[:,['continent']]

continent
0          Asia
1          Asia
2          Asia
3          Asia
4        Africa
...         ...
25475      Asia
25476      Asia
25477      Asia
25478      Asia
25479      Asia

[25480 rows x 1 columns]

**read the multiple columns**

- whenever we see multiple means recall the list
- keep the column in list
- pass the list

In [10]:
cols=['continent','case_status']
visa_df[cols]

continent case_status
0          Asia      Denied
1          Asia   Certified
2          Asia      Denied
3          Asia      Denied
4        Africa   Certified
...         ...         ...
25475      Asia   Certified
25476      Asia   Certified
25477      Asia   Certified
25478      Asia   Certified
25479      Asia   Certified

[25480 rows x 2 columns]

$task-1$

**we want to find the the number of application continent wise**

- there is as asia continent is there
- how many applicants are from Asia out of 25480

**unique**

- unique will provide how many unique items are there in a particular column

In [11]:
visa_df['continent'].unique()

array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
       'Oceania'], dtype=object)

In [12]:
len(visa_df['continent'].unique())

6

**nunique**

- gives count of unique items

In [13]:
visa_df['continent'].nunique()

6

In [15]:
# what is didderence between len(unique) and nunique
len(visa_df['continent'].unique())
visa_df['continent'].nunique()
# max()
# len()
# min()
# sum()

6

In [ ]:
con=visa_df['continent']=='Asia'
con

In [ ]:
# step-1: read the data with specific column
# step-2: apply the condition
#          when we apply the condition python will give True and false
# step-3: in order to retrive tyhe True data
#          dataframe_name[con]
visa_df['continent']
con=visa_df['continent']=='Asia'
len(visa_df[con])

In [ ]:
con=visa_df['continent']=='Asia'
# dataframename[condition]
len(visa_df[con])
len(visa_df['continent']=='Asia')

In [ ]:
con=visa_df['continent']=='Asia'
len(visa_df[con])
con=visa_df['continent']=='Africa'
len(visa_df[con])
con=visa_df['continent']=='North America'
len(visa_df[con])
con=visa_df['continent']=='Europe'
len(visa_df[con])
con=visa_df['continent']=='South America'
len(visa_df[con])
con=visa_df['continent']=='Oceania'
len(visa_df[con])
con=visa_df['continent']==i
len(visa_df[con])

In [ ]:
con=visa_df['continent']=='Asia'
len(visa_df[con])

con=visa_df['continent']=='Africa'
len(visa_df[con])

con=visa_df['continent']=='North America'
len(visa_df[con])

con=visa_df['continent']=='Europe'
len(visa_df[con])

con=visa_df['continent']=='South America'
len(visa_df[con])

con=visa_df['continent']=='Oceania'
len(visa_df[con])

con=visa_df['continent']==i
len(visa_df[con])

In [ ]:
for i in visa_df['continent'].unique():
    print(i)

In [ ]:
for i in visa_df['continent'].unique():
    con=visa_df['continent']==i
    count=len(visa_df[con])
    print(f"the number of application from {i} is: {count}")

$task-2$

- now instead of print count , take empty list

- Save the values in a empty list

- we already has unique continent list

- we also having count list

- Create dataframe

In [ ]:
continents=visa_df['continent'].unique()

count=[]
for i in visa_df['continent'].unique():
    con=visa_df['continent']==i
    count.append(len(visa_df[con]))

cols=['Continents','Nummber of Applicants']
pd.DataFrame(zip(continents,count),
            columns=cols)

In [ ]:
continents=visa_df['continent'].unique()

count=[]
for i in visa_df['continent'].unique():
    con=visa_df['continent']==i
    count.append(len(visa_df[con]))

cols=['Continents','Nummber of Applicants']
continents_df=pd.DataFrame(zip(continents,count),
                columns=cols)
continents_df.to_csv('continents_df.csv',index=False)

**value count**

In [ ]:
visa_df['continent'].value_counts()

- we created a dataframe using dataframe of unique of a columns and its count

- we created by our own way,by using for loop

- the same operation we can get using **value_count**

In [ ]:
cdf=visa_df['continent'].value_counts()
cdf

In [ ]:
dir(cdf)

In [ ]:
type(cdf)

In [ ]:
keys=cdf.keys()

In [ ]:
values=cdf.values

In [ ]:
pd.DataFrame(zip(keys,values))

In [ ]:
cdf=visa_df['continent'].value_counts()
keys=cdf.keys()
values=cdf.values
cols=['Lables','Count']
df=pd.DataFrame(zip(keys,values),
               columns=cols)
df.to_csv('Continent.csv',index=False)

In [ ]:
cat_columns[1:]

In [ ]:
cdf=visa_df['education_of_employee'].value_counts()
keys=cdf.keys()
values=cdf.values
cols=['Lables','Count']
df=pd.DataFrame(zip(keys,values),
               columns=cols)
df.to_csv('education_of_employee.csv',index=False)

- we need to change column name
- we need to change save the file name as column name

In [ ]:
for i in cat_columns[1:]:
    cdf=visa_df[i].value_counts()
    keys=cdf.keys()
    values=cdf.values
    cols=['Lables','Count']
    df=pd.DataFrame(zip(keys,values),
                   columns=cols)
    path=r"C:\Users\md mumtaz\Documents\Naresh it\EDA\visa_datafiles"
    df.to_csv(f'{path}\\{i}t.csv',index=False)

In [ ]:
def valuecounts(i):
    cdf=visa_df[i].value_counts()
    keys=cdf.keys()
    values=cdf.values
    cols=['Lables','Count']
    df=pd.DataFrame(zip(keys,values),
                   columns=cols)
    path=r"C:\Users\md mumtaz\Documents\Naresh it\Visadataset.csv"
    df.to_csv('Continent.csv',index=False)

In [ ]:
# visa_df[cat_columns[1:]].applymap(valuecounts)

**Normalize**

- Relative frequency table
- will give percentage of lables in the coulumn

In [ ]:
cdf=visa_df['continent'].value_counts(normalize=True)
round(cdf*100,2)

**Bar chart**

- Bar chart is a Grafical reprersentation of frequency table
- Bar graph is plot between count and class
- in order to plot bar chart we required two columns
  - Categorical column
  - Numerical column
- we alredy creatyed a frequency table for **continent_df**
  - here we have cols=['Continent','Number of apllications']
  - Categorical column is : **Continent**
  - Numerical column is: **Number of applicants**
- package: **matplotlib**
- we required a dataframe

In [ ]:
cdf=visa_df['continent'].value_counts(normalize=True)
cdf

In [ ]:
import matplotlib.pyplot as plt
plt.bar('Continents', # cat column x-axis
        'Nummber of Applicants',  # Num column:Height y-axis
        data=continents_df)

plt.title("Bar chart")
plt.xlabel('Continents')
plt.ylabel("Nummber of Applicants")
plt.savefig('Continents_bar_chart.jpg')

In [ ]:
continents_df

In [ ]:
visa_df['continent']  # raw data
# continent_df
# for loop
# value counts
# plot the bar chart:matplotlib

In [ ]:
continents_df

In [ ]:
############################################task-1###########################################
- take a fresh notbook
- read the visa dataset
- create a case_status frequency table
- save that in your laptop

take another noteboook 
- perform the bar pot on case_status

#####################################task-2###################################################
- take a fresh not book
- read the dataset
- create a case_status frequency table
- perform the bar pot on case_status datframe

In [ ]:
round(2.564,2)
